src/app/delegates/components/EndorsedFilter.tsx

```diff
@@ -0,0 +1,78 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import { Listbox } from "@headlessui/react";
+import { Fragment } from "react";
+import { ChevronDown } from "lucide-react";
+import { useAddSearchParam, useDeleteSearchParam } from "@/hooks";
+import { useAgoraContext } from "@/contexts/AgoraContext";
+
+const FILTER_PARAM = "endorsedFilter";
+const DEFAULT_FILTER = "all";
+
+export default function EndorsedFilter() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const addSearchParam = useAddSearchParam();
+  const deleteSearchParam = useDeleteSearchParam();
+  const filterParam = searchParams?.get(FILTER_PARAM) || "all";
+  const { setIsDelegatesFiltering } = useAgoraContext();
+
+  let endorsedFilterOptions: any = {
+    all: {
+      value: "All Delegates",
+      sort: "all",
+    },
+    true: {
+      value: "Endorsed Delegates",
+      sort: "true",
+    },
+  };
+
+  const handleChange = (value: string) => {
+    setIsDelegatesFiltering(true);
```

Comment:
@mcgingras this flag is used to display the "flashing" loading state while the list is being filtered. I will have to refactor this later.